design/electrical-system/README.md

```diff
@@ -1,19 +1,159 @@
+amenable to global supply chains, be low cost, and yet be robust and
+reliable.
 
-[Wiring](wiring.md)
+## Computing Architecture
 
-[PCB](../../pcb)
+A dual-computer architecture was selected.  There is a Cycle Controller
+based on STM32 that handles the real-time control of the ventilator’s
+sensors and actuators.  Connected to this is a UI Computer based on a
+Touchscreen, Raspberry Pi 4, and Linux to provide the user interface.
+ This design allows us to delegate the most critical functions to a
+simple design that can be reasonably reviewed and tested in its
+entirety, while also allowing enough complexity in the UI Computer to
+provide a comprehensive user interface.
 
+![](images/system_block_diagram.png)
 
-**TODO:** provide context for the following:
+The division of responsibility between the two computing elements is
+divided roughly by the immediacy of a potential failure.  For the class
+of operations in which a potential failure could result in a hazard too
+quickly for reasonable human intervention, the Cycle Controller is used.
+ For the class of operations in which a potential failure can reasonably
+be addressed by human interaction in a reasonable amount of time given a
+timely alarm, the UI Computer is used.  This design also allows for
+self-checking between the two computing elements to mitigate
+single-point failures.
 
-[Plotting and recording live data from MCU](data-from-mcu.md)
+The STM32 microcontroller was selected because it is more powerful than
+Arduinos or similar popular microcontrollers, but it is still cheap,
+widely available, well-documented, and has extensive debugging tools.
+The STM32 is proven to be a reliable medical electrical component and
+has a diverse I/O set for pneumatic and mechanical control.  The
+operation of the Cycle Controller is kept as simple as possible to
+facilitate comprehensive review, test, and verification of its function.
+
+The Raspberry Pi was selected as the UI controller for its wide
+compatibility with existing human-machine interfaces and robustness to
+supply chain shock. The Raspberry Pi also enables rapid software and UI
+development by supporting a Linux-based operating system and the common
+HDMI interface. It supports remote monitoring, as well as automated
+testing, through its wired and wireless network interfaces.  While
+powerful and flexible, the complexity of its operating system means that
+real-time control cannot be sufficiently guaranteed or verified.  Hence
+all time-critical operations, such as ventilator pneumatic control, are
+handled by the STM32.
+
+A serial communication link using checksums to ensure data integrity
+provides reliable communication of settings and data between the two
+computing elements.
+
+In the event of a system crash or failure in the UI Computer, the Cycle
+Controller will continue to operate the ventilator with the most recent
+valid settings received from the UI Computer.
```

Comment:
Though it makes sense to have some elements of software behaviour in the electrical design, I think this paragraph is a bit less relevant to electrical, as it describe something that is mostly software (admittedly enabled by a hardware choice).
I guess it feels redundant to me with the software architecture discussion, and we could just say the dual-computer design allows more flexibility and robustness in the software design, and link to the SW architecture file, to limit redundancy?

Reply:
Thanks. Good catch! Removed this as indeed it is redundant.